### PR TITLE
ESSNTL-3939: update the check of RHEL release

### PR DESCRIPTION
**Pull request by xiangce**
- the original checking of redhat_release is not sufficient
- involve the check of OSRelease combiner in this PR

### CODE REVIEW

**Comment by xiangce** [comment]
@r14chandra - please kindly review this PR whenever free. Thanks.

File: src/puptoo/process/profile.py [hunk 1]
```diff
@@ -392,19 +395,30 @@ def system_profile(
             catch_error("uname", e)
             raise
 
-    if redhat_release:
+    if redhat_release and os_release:
         try:
+            # The 'redhat_release.rhel' may be "Red Hat Enterprise Linux xxx"
             profile["os_release"] = redhat_release.rhel
```
Comment (r14chandra) [inline src/puptoo/process/profile.py:401]: So, are we populating `os_release` for other OS as well? I thought this block (i.e checking `redhat_release` is not null) is only for RHEL systems. Please correct me.

File: src/puptoo/process/profile.py [hunk 2]
```diff
@@ -392,19 +395,30 @@ def system_profile(
...
-            if profile.get("host_type") is None:
-                if redhat_release.major >= 8:
-                    profile["system_update_method"] = "dnf"
-                else:
-                    profile["system_update_method"] = "yum" 
+            # But it might not be an RHEL, see
+            # - https://github.com/RedHatInsights/insights-core/blob/master/insights/combiners/os_release.py
+            if os_release.is_rhel:
+                profile["operating_system"] = {
+                    "major": redhat_release.major,
```
Comment (r14chandra) [inline src/puptoo/process/profile.py:406]: Are these `redhat_release.major` and `redhat_release.minor` not provided for other OS?

File: src/puptoo/process/profile.py (hunk 1 above)
Comment (xiangce) [reply src/puptoo/process/profile.py:401]: I think so.
But here I just try my best to make sure that the new code stays compatible with the original code. 
The original code, in fact, didn't consider if the `redhat_release` is correct or not, and so does the new code.

From my perspective, I think we should update this line as 
```python
profile["os_release"] = os_release.release
```
which is a bit accurate and shows the `os_release` as `RHEL` OR `"Oracle", "Fedora"`, etc, when it's not an RHEL System. See [here](https://github.com/RedHatInsights/insights-core/blob/3.0/insights/combiners/os_release.py#L31)

File: src/puptoo/process/profile.py (hunk 2 above)
Comment (xiangce) [reply src/puptoo/process/profile.py:406]: Yep, for other OS, these two items look pointless, so I leave them as `-1` for NON-RHEL.

Comment (xiangce) [reply src/puptoo/process/profile.py:406]: BTW, just FYI,

According to the `advisor-backend` ([here is the code](https://gitlab.cee.redhat.com/insights-platform/advisor-backend/-/blob/master/service/service.py#L364)), and per my test: When a host is identified as a NON-RHEL via the [OSRelease](https://github.com/RedHatInsights/insights-core/blob/3.0/insights/combiners/os_release.py) combiner, only this [OTHER_LINUX_SYSTEM recommendations](https://console.redhat.com/insights/advisor/recommendations?rule_status=enabled&sort=-total_risk&limit=20&offset=0&text=OTHER_LINUX#SIDs=&tags=) will be shown to the customer on the Advisor UI.

And the update of this PR is trying to make sure the "General Information" ([here](https://console.redhat.com/insights/inventory/72bdbdec-40d6-48ee-934f-e9a80915982b)) on Advisor would be more accurate, even when it's a NON-RHEL.

File: src/puptoo/process/profile.py (hunk 1 above)
Comment (r14chandra) [reply src/puptoo/process/profile.py:401]: Confirmed. Yes, `operating_system` field can be for any OS - [inventory_schema](https://github.com/RedHatInsights/inventory-schemas/blob/master/tests/utils/valids.py#L80).
So does that mean, `redhat_release.rhel` (as per current code) can give us a release version of ANY OS? 
[.rhel](https://github.com/RedHatInsights/insights-core/blob/master/insights/parsers/redhat_release.py#L116) is not specific to RHEL?

File: src/puptoo/process/profile.py (hunk 2 above)
Comment (r14chandra) [reply src/puptoo/process/profile.py:406]: > Yep, for other OS, these two items look pointless, so I leave them as `-1` for NON-RHEL.

I see, in the test examples in [inventory_schema](https://github.com/RedHatInsights/inventory-schemas/blob/master/tests/utils/valids.py#L80), other OS major minor versions are valid. So, should we populate the fields for other OS as well instead of adding `-1`?

File: src/puptoo/process/profile.py (hunk 1 above)
Comment (xiangce) [reply src/puptoo/process/profile.py:401]: You are right. I made a mistake in the former comment, the `os_release` should not be `"RHEL"` OR `"Oracle", "Fedora"`, it should be the `release version of the OS`. Sorry about that.

The [redhat_release.rhel](https://github.com/RedHatInsights/insights-core/blob/master/insights/parsers/redhat_release.py#L116) tries to give the RHEL release version, just according to the literal text of the `/etc/redhat-release`, even when it isn't RHEL since the `/etc/redhat-release` can be mocked as RHEL freely (we do have such cases).

If here, we just want to show the release version ONLY for RHEL in this `os_release` field, we can do such change:
```python
if os_release.is_rhel:
   profile["os_release"] = redhat_release.rhel
else:
   profile["os_release"] = "null"   # or None
```
Does this work?

File: src/puptoo/process/profile.py (hunk 2 above)
Comment (xiangce) [reply src/puptoo/process/profile.py:406]: It's a bit complex..., here in this puptoo `profile` module, the `redhat_release.major/minor` is from the [RedHatReleasse Combiner](https://github.com/RedHatInsights/insights-core/blob/master/insights/combiners/redhat_release.py#L38-L51), this combiner tries to get the `major/minor` from the `Uname` parser first which depends on a hardcoded [mapping](https://github.com/RedHatInsights/insights-core/blob/master/insights/parsers/uname.py#L46-L119)  which is from this [public KCS](https://access.redhat.com/articles/3078), only when the `Uname` parser is not available or [parseable (-1, -1)](https://github.com/RedHatInsights/insights-core/blob/master/insights/parsers/uname.py#L370), the combiner [turns to check](https://github.com/RedHatInsights/insights-core/blob/master/insights/combiners/redhat_release.py#L73) the [RedhatRelease parser](https://github.com/RedHatInsights/insights-core/blob/master/insights/parsers/redhat_release.py#L35) which just read the literal text of the `etc/redhat-release`.

TLDR;, the `redhat_release.major/minor` is not accurate to indicate the release version, especially when it's other OS.  

For example: assume the host is an `CentOS Linux` but booted with a valid RHEL 7.6 kernel ([valid in this KCS](https://access.redhat.com/articles/3078)), assume its `etc/redhat-release` is `CentOS Linux 7.0`, in this case, the `major.minor` will be set to `7.6` but not `7.0`. the `7.6` is not correct for indicating the release version of CentOS and also not correct for RHEL since it will be identified as `os_release.is_rhel == False`.

**Comment by xiangce** [comment]
@r14chandra - yes, closed. Thanks for the reminder.
